CRM/Contribute/BAO/ManagePremiums.php

```diff
@@ -127,16 +120,16 @@ public static function del($productID) {
     $premiumsProduct = new CRM_Contribute_DAO_PremiumsProduct();
     $premiumsProduct->product_id = $productID;
     if ($premiumsProduct->find(TRUE)) {
-      $session = CRM_Core_Session::singleton();
-      $message .= ts('This Premium is being linked to <a href=\'%1\'>Online Contribution page</a>. Please remove it in order to delete this Premium.', array(1 => CRM_Utils_System::url('civicrm/admin/contribute', 'reset=1')), ts('Deletion Error'), 'error');
-      CRM_Core_Session::setStatus($message);
-      return CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/admin/contribute/managePremiums', 'reset=1&action=browse'));
+      $message = ts('This Premium is being linked to <a href=\'%1\'>Online Contribution page</a>. Please remove it in order to delete this Premium.', array(1 => CRM_Utils_System::url('civicrm/admin/contribute', 'reset=1')));
+      CRM_Core_Session::setStatus($message, ts('Deletion Error'), 'error');
+      CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/admin/contribute/managePremiums', 'reset=1&action=browse'));
+    }
+    else {
+      //delete from financial Type table
+      $premium = new CRM_Contribute_DAO_Product();
+      $premium->id = $productID;
+      $premium->delete();
     }
-
-    //delete from financial Type table
-    $premium = new CRM_Contribute_DAO_Product();
-    $premium->id = $productID;
-    $premium->delete();
```

Comment:
For someone skimming, this change looks bigger than it is -- basically it seems to:

* Change an incorrect `.=` to `=`
* Make the `if/else` logic more visible/transparent.

Reply:
Correct, as `CRM_Utils_System::redirect` does not return a value it was throwing warnings in the IDE (PHPStorm).  But removing "return" makes the code less clear, so the else is added to make it clear again.
And the parameters to setup the status message were a mess, so that's fixed too.